articles/152_roslaunch_multi_machine_launch.md

```diff
@@ -0,0 +1,310 @@
+                        OVERRIDE_LAUNCH_PROCESS_OUTPUT envvar.
+```
+
+Example output:
+
+```bash
+$ ros2 launcher launch demo_nodes_cpp talker_listener.launch.py
+[INFO] [launch]: All log files can be found below /home/preed/.ros/log/2019-09-11-20-54-30-715383-regulus-2799
+[INFO] [launch]: Default logging verbosity is set to INFO
+[INFO] [launch]: Launch System ID is 50bda6fb-d451-4d53-8a2b-e8fcdce8170b
+[INFO] [talker-1]: process started with pid [2809]
+[INFO] [listener-2]: process started with pid [2810]
+[talker-1] [INFO] [talker]: Publishing: 'Hello World: 1'
+[listener-2] [INFO] [listener]: I heard: [Hello World: 1]
+[talker-1] [INFO] [talker]: Publishing: 'Hello World: 2'
+[listener-2] [INFO] [listener]: I heard: [Hello World: 2]
+[talker-1] [INFO] [talker]: Publishing: 'Hello World: 3'
+[listener-2] [INFO] [listener]: I heard: [Hello World: 3]
+[talker-1] [INFO] [talker]: Publishing: 'Hello World: 4'
+[listener-2] [INFO] [listener]: I heard: [Hello World: 4]
+^C[WARNING] [launch]: user interrupted with ctrl-c (SIGINT)
+[listener-2] [INFO] [rclcpp]: signal_handler(signal_value=2)
+[INFO] [talker-1]: process has finished cleanly [pid 2809]
+[INFO] [listener-2]: process has finished cleanly [pid 2810]
+[talker-1] [INFO] [rclcpp]: signal_handler(signal_value=2)
+```
+
+Note how there is one difference from the old behavior of `ros2 launch`; the
+group of nodes is assigned a Launch System ID.  This is a unique identifier
+that can be used to track all of the nodes launched by a particular command
+across a network.
+
+Additionally, it is possible to detach from a system and let it run in the
+background:
+
+```bash
+$ ros2 launcher launch -D demo_nodes_cpp talker_listener.launch.py
+[INFO] [launch]: All log files can be found below /home/preed/.ros/log/2019-09-11-20-54-30-715383-regulus-2799
+[INFO] [launch]: Default logging verbosity is set to INFO
+[INFO] [launch]: Launch System ID is 50bda6fb-d451-4d53-8a2b-e8fcdce8170b
+$
+```
+
+#### `list`
+
+Since it is possible to launch a system of nodes that spans a network and detach
+from it, it is necessary to be able to query the network to find which systems
+are active.
+
+```bash
+$ ros2 launcher list -h
+usage: ros2 launcher list [-h] [-v] [--spin-time SPIN_TIME]
+
+List running launch systems
+
+
+optional arguments:
+  -h, --help            Show this help message and exit.
+  -v, --verbose         Provides more verbose output.
+  --spin-time SPIN_TIME
+                        Spin time in seconds to wait for discovery (only
+                        applies when not using an already running daemon)
+```
+
+Example output:
+
+```bash
+$ ros2 launcher list
+ab1e0138-bb22-4ec9-a590-cf377de42d0f
+50bda6fb-d451-4d53-8a2b-e8fcdce8170b
+5d186778-1f50-4828-9425-64cc2ed1342c
+$
+```
+
+```bash
+$ ros2 launcher list
+ab1e0138-bb22-4ec9-a590-cf377de42d0f: 5 nodes, 2 hosts
+50bda6fb-d451-4d53-8a2b-e8fcdce8170b: 2 nodes, 1 host
+5d186778-1f50-4828-9425-64cc2ed1342c: 16 nodes, 3 hosts
```

Comment:
@mlanting does each launch file provide enough information to dispatch each node to the right host? If so, how does this cope with a single-machine launch to be launched in multiple hosts, unaware of each other? If not, how are nodes or even "systems" as you propose here associated with each host?

Reply:
This is just a theoretical design of how we would expect the CLI to look like, so there isn't a definite answer for those questions (yet).  I expect that the launch files will, similar to ROS1's launch files, contain information about all of the hosts involved and which nodes should be launched on which host.

> If so, how does this cope with a single-machine launch to be launched in multiple hosts, unaware of each other?
I'm not sure what this means, do you have an example?

Reply:
My thought was that nodes which need to run on a certain machine can have that machine specified in the launch file, and those that do not have a machine specified would be sent to a host selected by the launch system (to allow for things like load balancing). The launch system would be informed of hosts by something like a "declare_launch_hosts_action"